pkg/piecestore/rpc/client/client.go

```diff
@@ -84,27 +89,33 @@ func NewCustomRoute(route pb.PieceStoreRoutesClient, nodeID *node.ID, bandwidthM
 		bandwidthMsgSize = *defaultBandwidthMsgSize
 	}
 
-	return &Client{
-		route:            route,
+	return &Piecestore{
+		client:           client,
 		bandwidthMsgSize: bandwidthMsgSize,
 		prikey:           prikey,
-		nodeID:           nodeID,
+		nodeID:           node.IDFromString(_node.GetId()),
 	}, nil
 }
 
 // Close closes the connection with piecestore
-func (client *Client) Close() error {
-	return client.conn.Close()
+func (ps *Piecestore) Close() error {
+	if ps.closeFunc == nil {
+		// TODO@bryanchrswhite/aleitner: 	This will happen if someone uses `NewCustomRoute`.
+		// 																What is the expected behavior?
```

Comment:
`NewCustomRoute` is used by the tests. So in this case the connection would be mocked, so returning just `nil` would be fine.